packages/core/src/crypto/x509/X509Service.ts

```diff
@@ -0,0 +1,60 @@
+import * as x509 from '@peculiar/x509'
+import { injectable } from 'tsyringe'
+
+import { AgentContext } from '../../agent'
+import { WebCrypto } from '../webcrypto'
+
+import { X509Certificate } from './X509Certificate'
+import { X509Error } from './X509Error'
+
+@injectable()
+export class X509Service {
+  public webCrypto: WebCrypto
+
+  public constructor(agentContext: AgentContext) {
+    this.webCrypto = new WebCrypto(agentContext)
+    x509.cryptoProvider.set(this.webCrypto)
```

Comment:
If we pass the `agentContext` into every method it has, yes this is easily doable!

Reply:
Also, this is in another case the wrong place. If we want to reuse the webcrypto in other services we should register it at a higher level. But I will pass the `agentContext` to each service call and inject it there into the method.

Reply:
Okay good. and it won't overwrite it for the whole x509 package? As the current method sets the crypto provider globally right?

Reply:
Right now, in the new implementation I set it and clear it. AFAIK its a local global, not `global.crypto` directly.

Reply:
Yes but that could still cause conflicts with async methods doing multiple operations right?

Because if method X is called twice concurrently. After doing one operation, method x needs to wait for an async operation, and then the other x method is ran. Now the x509 crypto will be overriden. And when the first call to X resumes it will have another crypto instance, possibly bound to another agent context?

Or are all calls to x509 library sync and do we set it before each call to x509 library?